geowebcache/diskquota/core/src/main/java/org/geowebcache/diskquota/ConfigLoader.java

```diff
@@ -120,52 +138,61 @@ public ConfigLoader(final DefaultStorageFinder storageFinder,
      * @throws ConfigurationException
      */
     public void saveConfig(DiskQuotaConfig config) throws IOException, ConfigurationException {
-        File rootCacheDir = getRootCacheDir();
+        if (!resourceProvider.hasOutput()) {
+            log.error("Unable to save DiskQuota to resource :" + resourceProvider.getLocation());
+            return;
+        }
+        
         XStream xStream = getConfiguredXStream(new GeoWebCacheXStream());
-        final File configFile = new File(rootCacheDir, CONFIGURATION_FILE_NAME);
-        final File tmpConfigFile = new File(rootCacheDir, CONFIGURATION_FILE_NAME + ".tmp");
-        log.debug("Saving disk quota config to " + configFile.getAbsolutePath());
-        OutputStream configOut = new FileOutputStream(tmpConfigFile);
+        
+        log.debug("Saving disk quota config to " + resourceProvider.getLocation());
+        OutputStream configOut = resourceProvider.out();
         try {
             xStream.toXML(config, new OutputStreamWriter(configOut, "UTF-8"));
         } catch (RuntimeException e) {
-            log.error("Error saving DiskQuota config to temp file :"
-                    + tmpConfigFile.getAbsolutePath());
+            log.error("Error saving DiskQuota config to file :"
+                    + resourceProvider.getLocation());
         } finally {
             configOut.close();
         }
-        configFile.delete();
-        if (!FileUtils.renameFile(tmpConfigFile, configFile)) {
-            throw new ConfigurationException("Couldn't save disk quota config file "
-                    + configFile.getAbsolutePath());
-        }
     }
 
     public DiskQuotaConfig loadConfig() throws IOException, ConfigurationException {
         DiskQuotaConfig quotaConfig;
-        final File configFile = getConfigResource();
-        if (!configFile.exists()) {
-            log.info("DiskQuota configuration not found: " + configFile.getAbsolutePath());
-            quotaConfig = new DiskQuotaConfig();
+        InputStream configIn;
```

Comment:
try-with-resources variables are implicitly final so, if used here, you might consider refactoring finding the config location into a helper method.